src/Reflection/ReflectionClassConstant.php

```diff
@@ -0,0 +1,202 @@
+<?php
+
+namespace Roave\BetterReflection\Reflection;
+
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\ClassConst;
+use Roave\BetterReflection\NodeCompiler\CompileNodeToValue;
+use Roave\BetterReflection\NodeCompiler\CompilerContext;
+use Roave\BetterReflection\Reflector\Reflector;
+
+class ReflectionClassConstant implements \Reflector
+{
+
+    /**
+     * @var bool value of const was cached?
+     */
+    private $valueCached = false;
+
+    /**
+     * @var mixed const value
+     */
+    private $value;
+
+    /**
+     * @var
+     */
+    private $reflector;
+    /**
+     * @var ReflectionClass Constant owner
+     */
+    private $owner;
+
+    /**
+     * @var ClassConst
+     */
+    private $node;
+
+    private function __construct()
+    {
+    }
+
+    /**
+     * Create a reflection of a class's constant by Const Node
+     *
+     * @param Reflector $reflector
+     * @param ClassConst $node
+     * @param ReflectionClass $owner
+     * @return ReflectionClassConstant
+     */
+    public static function createFromNode(
+        Reflector $reflector,
+        ClassConst $node,
+        ReflectionClass $owner
+    ) : self {
+        $ref = new self();
+        $ref->node = $node;
+        $ref->owner = $owner;
+        $ref->reflector = $reflector;
+        return $ref;
+    }
+
+    /**
+     * Get the name of the reflection (e.g. if this is a ReflectionClass this
+     * will be the class name).
+     *
+     * @return string
+     */
+    public function getName() : string
+    {
+        return $this->node->consts[0]->name;
+    }
+
+    /**
+     * Returns constant value
+     *
+     * @return mixed
+     */
+    public function getValue()
+    {
+        if (false === $this->valueCached) {
```

Comment:
Consider flipping over the conditional, so the processing comes after an early return